tests/ActiveDataProviderTest.php

```diff
@@ -50,13 +50,15 @@ public function testQuery()
         $query->from('customer');
 
         $provider = new ActiveDataProvider([
+            'connectionClass' => 'yii\mongodb\Connection',
             'query' => $query,
             'db' => $this->getConnection(),
```

Comment:
Can't we determine connection from what's in db if it's specified?

Reply:
You mean like `get_class()`? What is the purpose?

Reply:
Not to specify a class if we have a class already?

Reply:
If I understand correctly - you would like `yii\data\ActiveDataProvider` to skip `ensure` checks on object-like `db`? Still strings and arrays need to be checked.